Put spotify permissions in props as well

paramsingh opened the pull request:
<!--
    Hello! Thanks for submitting a pull request to ListenBrainz. We appreciate
    your time and interest in helping our project!

    Use this template to help us review your change. Not everything is required,
    depending on your change. Keep or delete what is relevant for your change.
    Remember that it helps us review if you give more helpful info for us to
    understand your change.

    Ensure that you've read through and followed the Contributing Guidelines, in
    ./github/CONTRIBUTING.md.
-->

# Summary

* This is a…
    * ( ) Bug fix
    * (x) Feature addition
    * ( ) Refactoring
    * ( ) Minor / simple change (like a typo)
    * ( ) Other


# Problem

<!--
    Anything that helps us understand why you are making this change goes here.
    What problem are you trying to fix? What does this change address?
-->

* JIRA ticket (_optional_): <!-- [LB-XXX](https://tickets.metabrainz.org/browse/LB-XXX) -->

We had to make requests to the spotify API to check if streaming worked in #558 .

# Solution

<!--
    The details of your change. Talk about technical details, considerations, or
    other interesting points. If you have a lot to say, be more detailed in this
    section.
-->
Return the permissions we stored in props and the refresh-token endpoint.

MonkeyDo commented on listenbrainz/webserver/views/follow.py:
```diff
@@ -39,12 +40,13 @@ def follow(user_list):
         "follow_list": follow_list_members,
-        "spotify_access_token": spotify_access_token,
+        "spotify_access_token": spotify_data.get("access_token", ""),
+        "spotify_permission": spotify_data.get("permission", ""),
```
At this point, maybe we should consider returning an object for spotify props in all the views?  I.E:
```
"spotify": {
    "access_token": XYZ,
    "permissions":XYZ
}
```

It'll be a bit less polluting and better match the refresh-token response structure.

paramsingh replied:
Makes sense, I didn't want to break existing stuff which is why I didn't change it. I'll do it.

MonkeyDo commented on listenbrainz/webserver/static/js/jsx/profile.jsx:
```diff
@@ -352,13 +352,13 @@ class RecentListens extends React.Component {
                 listens={spotifyListens}
                 direction={this.state.direction}
-                spotify_access_token={this.props.spotify_access_token}
+                spotify={this.props.spotify}
```
Maybe rename that prop to `spotify_user` for clarity ?

paramsingh commented:
I've incorporated the changes in #558 here. Added a new function for permission errors because we don't want a popup on every page for users who don't want to listen to music.
